### DISCUSSION TITLE
Welcome to drug_named_entity_recognition Discussions!

### DISCUSSION THREAD

**Discussion by woodthom2** [fastdatascience/drug_named_entity_recognition#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by LKB159** [comment]
Hello, 
I am an analyst intern. I just started using github. The drug_named_entity_recognition library is amazing! I thought I would have to build something like this by hand. I did run into a problem. My text column has a lot of misspellings and aberrations of drug names. Has any one figured out a way to clean up a long text column before applying the drug_named_entity_recognition library? (excluding chatgpt) 
Let me know :)

**Comment by woodthom2** [comment]
Hi
Thanks so much for your message - this is really interesting to read about your use case.
Do you have a column of data which is known to be drug names, or are they sentences containing drug names ("I would usually prescribe 5mg diazepam for pts")?

As it currently stands the code doesn't handle misspellings but we could modify it to do so. 
One way is to index n-grams (e.g. trigrams) of each drug name. Then we would have some error tolerance. 
Do you feel able to try this and make a PR?

**Comment by austin-hwang** [comment]
Hi, I have a similar use case where I have a transcript with some mispelled prescription drugs which I want to identify. I saw in the README that fuzzy matching would be supported in version 2. Do you have a timeline on when it will be released? Thank you!

**Comment by woodthom2** [comment]
Hi Austin,
I'm testing the version with fuzzy matching at the moment, so I didn't put
it on Pypi yet. But if you clone the repository and cd into the base folder
of the repository and type "pip install ." in the console (aka terminal)
then you will have version 2 working - let me know please if this isn't
clear.
Thomas

**Comment by woodthom2** [comment]
I expect I can put it on Pypi in the next few weeks

**Reply by austin-hwang** [reply]
Thanks, this is very helpful. I'll test it locally as you mentioned first.

**Comment by woodthom2** [comment]
Thanks. I'd appreciate any feedback on any issues you encounter, then we
can fix them before I release the next version to Pypi

**Comment by woodthom2** [comment]
@LKB159 this might be interesting for your use case too (I appreciate you
opened the thread quite some time ago)
